tests/console/login.go

```diff
@@ -149,7 +149,7 @@ func LoginToFedora(vmi *v1.VirtualMachineInstance) error {
 				Rt: 10,
 			},
 			&expect.Case{
-				R: regexp.MustCompile(fmt.Sprintf(`\[fedora@%s ~\]\$ `, vmi.Name)),
+				R: regexp.MustCompile(fmt.Sprintf(`\[fedora@(localhost|%s) ~\]\$ `, vmi.Name)),
```

Comment:
I agree the fix is correct. Without it, vms with no dhcp server on the virt-launcher pod will fail login.
I wonder how it worked till now and what in this PR caused this issue to be visible?

Reply:
I will check the original fedora image to see if this is correct, but my current theory is that the unmodified fedora cloud-image has `fedora` set as hostname by default, while we we change this in the fedora with tooling image.

Reply:
But the name of the vmi in the tests is `testvmi_randnum`, so even if the default hostname is `fedora` is shouldn't work.

Reply:
That is indeed correct :/

Reply:
maybe the login is faster now, and since it managed to login in the first cycle
https://github.com/kubevirt/kubevirt/blob/a10e0328f83f66a87c13a1421aaedc3aea4babc4/tests/console/login.go#L159
it won't manage to relogin in the 2nd  ?

does the VM  at the end do have the hostname ?
if so it might make this theory stronger
hope i dont mix stuff

Reply:
I did not figure it out yet. We retry the login, but on the retry we do not check if we are already logged in. Therefore I am not sure how it would have managed in the past to see on the second try that it is already logged in. @oshoval but maybe that is not what you meant.

Reply:
https://github.com/kubevirt/kubevirt/blob/a10e0328f83f66a87c13a1421aaedc3aea4babc4/tests/console/login.go#L135
Here it already allowed localhost, and we see that we are logged in actually at the end of the 2nd retry
```
18:17:00: {"component":"tests","kind":"VirtualMachineInstance","level":"error","msg":"Retried login attempt after two minutes failed: [{Idx:2 Output:\r\n[fedora@localhost ~]$ \r\n[fedora@localhost ~]$  Match:[]}]","name":"testvmi-jqlj5","namespace":"kubevirt-test-default1","pos":"login.go:165","reason":"expect: timer expired after 60 seconds","timestamp":"2021-07-27T18:16:36.430114Z","uid":"bc2c3013-2a62-42a4-9001-5e8be59ed696"}
```

Maybe the fact that its asymetric and that the hostname was changed in the middle caused it, and it might return back later to be the real host name ?
the runcmd does set empty hostname
https://github.com/kubevirt/kubevirtci/pull/648/files#diff-60e708f2e40cf0f5d49420a079fdd43d2a2b2691cada7e745ee1033d195d8a80R43

Reply:
I am not sure. The point which @AlonaKaplan was making is that this tests starts only with the bridge network attached for which no dhcp server is started. Since the bridge network is attached, the pod network will not be attached. Therefore there is also no dhcp server from that side. Therefore the hostname namtching the VMI name should never have been offered and therfore it is not clear how this could pass.